index.js

```diff
@@ -33,176 +33,383 @@ const app = express()
+  const latestXethru = moment(XeThruData.timestamp, 'x')
+  const XeThruDelayMillis = currentTime.diff(latestXethru)
+
+  if (XeThruDelayMillis > XETHRU_THRESHOLD_MILLIS && !location.xethru_sent_alerts) {
+    helpers.log(`XeThru Heartbeat threshold exceeded; sending alerts for ${location.locationid}`)
+    await db.updateSentAlerts(location.locationid, true)
+    sendAlerts(location.locationid)
+  } else if (XeThruDelayMillis < XETHRU_THRESHOLD_MILLIS && location.xethru_sent_alerts) {
+    helpers.log(`XeThru at ${location.locationid} reconnected`)
+    await db.updateSentAlerts(location.locationid, false)
+    sendReconnectionMessage(location.locationid)
+  }
+}
+
+async function handleSensorRequest(currentLocationId) {
+  const statemachine = new StateMachine(currentLocationId)
+  const currentState = await statemachine.getNextState(db, redis)
+  const stateobject = await redis.getLatestLocationStatesData(currentLocationId)
+  let prevState
+  if (!stateobject) {
+    prevState = STATE.RESET
+  } else {
+    prevState = stateobject.state
+  }
+  const location = await db.getLocationData(currentLocationId)
+
+  // Check the XeThru Heartbeat
+  if (!helpers.isTestEnvironment()) {
+    await checkHeartbeat(currentLocationId)
+  }
+  helpers.log(`${currentLocationId}: ${currentState}`)
+
+  // Get current time to compare to the session's start time
+  const location_start_time = start_times[currentLocationId]
+  let sessionDuration
+
+  if (location_start_time !== null && location_start_time !== undefined) {
+    const start_time_sesh = new Date(location_start_time)
+    const now = new Date()
+
+    // Current Session duration so far:
+    sessionDuration = (now - start_time_sesh) / 1000
+  }
+
+  // If session duration is longer than the threshold (20 min), reset the session at this location, send an alert to notify as well.
+
+  if (sessionDuration * 1000 > location.auto_reset_threshold) {
+    autoResetSession(location.locationid)
+    start_times[currentLocationId] = null
+    helpers.log('autoResetSession has been called')
+    sendResetAlert(location.locationid)
+  }
+
+  helpers.log(`${sessionDuration}`)
+
+  // To avoid filling the DB with repeated states in a row.
+  // eslint-disable-next-line eqeqeq
+  if (currentState != prevState) {
+    await redis.addStateMachineData(currentState, currentLocationId)
+
+    // Checks if current state belongs to voidStates
+    if (VOIDSTATES.includes(currentState)) {
+      const latestSession = await db.getMostRecentSession(currentLocationId)
+
+      if (typeof latestSession !== 'undefined') {
+        // Checks if no session exists for this location yet.
+        if (latestSession.end_time == null) {
+          // Checks if session is open.
+          await db.updateSessionState(latestSession.sessionid, currentState)
+        }
+      }
+    }
+
+    // Checks if current state belongs to the session triggerStates
+    else if (TRIGGERSTATES.includes(currentState)) {
```

Comment:
I'm not a huge fan of how this put the `else` a couple lines after the closing brace on the `if`. Did it not like it when the line started with `}`? Or was this something that was there before?

Reply:
I am 99% sure it's because of the comment line, it doesn't like them wedged in-between things. I could make them in-line comments maybe?

Reply:
It doesn't like them on the same line so the options left are to add a `// prettier-ignore` but that goes above the line so I think it'd just cause the same problem since that's also a comment, or to put the explanation inside the block like this (it actually already is that way above, which I included):

```
      if (typeof latestSession !== 'undefined') {
        // Checks if no session exists for this location yet.
        if (latestSession.end_time == null) {
          // Checks if session is open.
          await db.updateSessionState(latestSession.sessionid, currentState)
        }
      }
    } else if (TRIGGERSTATES.includes(currentState)) {
      // Checks if current state belongs to the session triggerStates
      const client = await db.beginTransaction()
      const latestSession = await db.getMostRecentSession(currentLocationId, client)
```

Reply:
It's actually not a very useful comment. Let's get rid of it. Same with `Checks if current state belongs to voidStates` and `Checks if current state belongs to the session closingStates`

Reply:
Works for me! 😅